provider/provider.go

```diff
@@ -163,7 +199,7 @@ func (p *Provider) Provide(ctx context.Context, from, irreversibleFrom uint32, e
 				sort.Slice(nums, func(i, j int) bool { return nums[i] < nums[j] })
 
 				for _, num := range nums {
-					p.CurrentBlockNum = num
+					atomic.StoreUint32(&p.currentBlockNum, num)
```

Comment:
isnt event-provider synced? why do we need atomic here?

Reply:
Yes, it's not concurrent